mlflow/types/llm.py

```diff
@@ -0,0 +1,306 @@
+import time
+from dataclasses import asdict, dataclass, field
+from typing import List, Literal, Optional
+
+from mlflow.types.schema import Array, ColSpec, DataType, Object, Property, Schema
+
+# TODO: Switch to pydantic in a future version of MLflow.
+#       For now, to prevent adding pydantic as a core dependency,
+#       we use dataclasses instead.
+#
+#       Unfortunately, validation for generic types is not that
+#       straightforward. For example, `isinstance(thing, List[T])``
+#       is not supported, so the code here is a little ugly.
+
+
+class _BaseDataclass:
```

Comment:
all this validation logic is mainly to support the output validation done [here](https://github.com/mlflow/mlflow/pull/10820/files#diff-592813f9fb198dda0982794f2221a408f316709190b3bf02c0e77eb2c1107075R2023-R2024).

input validation shouldn't really be an issue, because it's handled by signature validation.

Reply:
Pydantic has flexible validation capability via [Validators](https://docs.pydantic.dev/latest/concepts/validators/), which may help reducing some boilorplate code. Have you given a try to it?